kakoune: Update plugins and fix powerline plugin

Flakebi opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
This fixes the powerline kakoune plugin, which was intermediately maintained at `jdugan6240/powerline.kak` but now moved back to `andreyorst/powerline.kak` (see also https://github.com/NixOS/nixpkgs/pull/108363#issuecomment-799800482).

Also update all other plugins and apply moved repositories (I didn’t know the update script has all these features, they are quite nice 😄).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @rmcgibbo @nrdxp

Flakebi committed:
kakounePlugins: update

Flakebi committed:
kakounePlugins: resolve github repository redirects

r-rmcgibbo commented:
Result of `nixpkgs-review pr 116485` at d8bd7816 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/99d192cb68c32f7015f4c7828a835e10">kakounePlugins.auto-pairs-kak</a></li></li>
    <li><a href="https://gist.github.com/0c22aa1601deaafaa4c0585435869040">kakounePlugins.connect-kak</a></li></li>
    <li><a href="https://gist.github.com/a4a2a72b966079b1dc55b32a06002f89">kakounePlugins.fzf-kak</a></li></li>
    <li><a href="https://gist.github.com/552743bc1879922cfabb54a474a628d9">kakounePlugins.powerline-kak</a></li></li>
    <li><a href="https://gist.github.com/e6f925521b23fb42bec3f1b4d08daa90">kakounePlugins.prelude-kak</a></li></li>
  </ul>
</details>
<details>
  <summary>6 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:171:5:
```
    |
171 |     meta.homepage = "https://github.com/kakounedotcom/prelude.kak/";
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:27:5:
```
   |
27 |     meta.homepage = "https://github.com/alexherbo2/auto-pairs.kak/";
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:39:5:
```
   |
39 |     meta.homepage = "https://github.com/kakounedotcom/connect.kak/";
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:171:5:
```
    |
171 |     meta.homepage = "https://github.com/kakounedotcom/prelude.kak/";
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:27:5:
```
   |
27 |     meta.homepage = "https://github.com/alexherbo2/auto-pairs.kak/";
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:39:5:
```
   |
39 |     meta.homepage = "https://github.com/kakounedotcom/connect.kak/";
   |     ^
```
</li>
  </ul>
</details>

Flakebi commented:
Take out the update of `auto-pairs-kak`, the updated `auto-pairs-kak` needs #115011, otherwise it fails to load.

nrdxp commented:
is there something wrong with kak-buffers also? I am an error about no `info-buffers` command after upgrading to the previous PR.

SuperSandro2000 commented on pkgs/applications/editors/kakoune/plugins/deprecated.json:
```diff
@@ -1 +1 @@
-{}
+{}
```
Please add a final new line back.

Flakebi replied:
Added again, thanks.
Also fixed the script to add the newline automatically.

Flakebi committed:
scripts.pluginupdate: write final newline

json.dump does not write any newline, so the final newline in the file
is missing.

nrdxp commented:
Perhaps other actively maintained [plugins](https://kakoune.org/plugins.html) could also be added since doing so is trivial. I'd like to see this in good shape in time for release with as many useful plugins as possible.

Essentially, my reasoning is that getting a PR merged takes some time and effort, and I'd hate to see authors go through that cycle again and again just to add a single kakoune plugin.
